src/components/fields/Percentage/Settings.tsx

```diff
@@ -0,0 +1,186 @@
+import { useEffect, useState } from "react";
+
+import {
+  Box,
+  Checkbox,
+  Grid,
+  InputLabel,
+  MenuItem,
+  TextField,
+  Typography,
+  useTheme,
+} from "@mui/material";
+import ColorPickerInput from "@src/components/ColorPickerInput";
+import { ISettingsProps } from "@src/components/fields/types";
+
+import { Color, toColor } from "react-color-palette";
+import { fieldSx } from "@src/components/SideDrawer/utils";
+import { resultColorsScale, defaultColors } from "@src/utils/color";
+import { GlobalStyles } from "tss-react";
+
+const colorLabels: { [key: string]: string } = {
+  0: "Start",
+  1: "Middle",
+  2: "End",
+};
+
+export default function Settings({ onChange, config }: ISettingsProps) {
+  const [colorsDraft, setColorsDraft] = useState<any[]>(
+    config.colors ? config.colors : defaultColors
+  );
+
+  const [checkStates, setCheckStates] = useState<boolean[]>([
+    colorsDraft[0],
+    colorsDraft[1],
+    colorsDraft[2],
+  ]);
+
+  useEffect(() => {
+    onChange("colors")(colorsDraft);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [colorsDraft]);
+
+  const onCheckboxChange = (index: number, checked: boolean) => {
+    setColorsDraft(
+      colorsDraft.map((value: any, idx: number) =>
+        index === idx ? (checked ? value || defaultColors[idx] : null) : value
+      )
+    );
+    setCheckStates(
+      checkStates.map((value, idx) => (index === idx ? checked : value))
+    );
+  };
+
+  const handleColorChange = (index: number, color: Color): void => {
+    setColorsDraft(
+      colorsDraft.map((value, idx) => (index === idx ? color.hex : value))
+    );
+  };
+
+  return (
+    <>
+      <GlobalStyles
+        styles={{
+          "& .MuiList-root.MuiMenu-list": {
+            padding: 0,
+          },
+        }}
+      />
+      <Grid container>
+        {checkStates.map((checked: boolean, index: number) => {
+          const colorHex = colorsDraft[index];
+          return (
+            <Grid
+              xs={12}
+              md={4}
+              item
+              sx={{
+                display: "flex",
+                alignItems: "end",
+                justifyContent: "center",
+              }}
+            >
+              <Checkbox
+                checked={checked}
+                sx={[
+                  fieldSx,
+                  {
+                    width: "auto",
+                    boxShadow: "none",
+                    backgroundColor: "inherit",
+                    "&:hover": {
+                      backgroundColor: "inherit",
+                    },
+                  },
+                ]}
+                onChange={() => onCheckboxChange(index, !checked)}
+              />
+              <TextField
+                select
+                label={colorLabels[index]}
+                value={1}
+                fullWidth
+                disabled={!checkStates[index]}
```

Comment:
```suggestion
                disabled={!checkStates[index]}
                sx={{
                  "& .MuiList-root.MuiMenu-list": {
                    padding: 0,
                  },
                }}
```

Moved the styles out of GlobalStyles above

Reply:
This acts as a modal and is opened in the presentation div in the root. 

It has 4px padding on my screen. ColorPicker has 12px(spacing(1.5)). So, paddingTop(1) solves this in my practice. I'm just not sure about if MuiList gets as same as spacing(0.5) which is 4px here, right?